### PR TITLE
feat(timm_seresnet): add timm SE-ResNet image-classification family

### PR DESCRIPTION
**Pull request by zhenshanx-nv**
## Background

`timm_resnet` (#1121) covers the gate-free ResNet and ResNeXt variants. The
squeeze-excite variants are a separate timm architecture family and are not
supported: `timm/seresnet50.a1_in1k` cannot be built or served today.

## Exit Criteria

- A `timm_seresnet` family builds timm SE-ResNet checkpoints from HF-hosted
  safetensors and produces logits matching timm's own implementation.
- The family's prefixes are disjoint from `timm_resnet`, so no checkpoint is
  claimed by both.
- The family is registered across the runtime strategy matrix, validation
  workloads, benchmark suite, website data, and the E2E model registry.

Non-goals: quantized builds and tensor-parallel builds.

## Implementation

The stage and block layout is recovered from the checkpoint exactly as in
`timm_resnet`: block counts from the `layerN.M` indices, basic versus bottleneck
from the presence of `conv3`, and the group count from the `conv2` input-channel
ratio.

Each block carries a squeeze-excite gate applied to its output **after the final
batch norm and before the residual add**. The gate uses a ReLU inner activation
with a plain sigmoid; timm names its two 1x1 projections `fc1` and `fc2`.

The gate is **required**, not optional. A checkpoint without one is rejected
rather than built as a plain ResNet, which would produce correct shapes and
wrong numbers. `timm_resnet` matches `resnet`/`resnext`/`wide_resnet` while this
family matches `seresnet`/`seresnext`/`legacy_seresnet`, so the two never
compete for the same checkpoint.

No public API, ABI, or bundle format change. No new dependencies.

### Change categories

- [x] Model or runtime behavior
- [ ] Public API
- [ ] ABI
- [ ] Bundle or artifact format
- [ ] Dependencies
- [ ] Documentation only
- [ ] CI or developer tooling

### Commands and Results

```
python -m pytest tests/builder/ tests/tools/ tests/e2e_harness/ -q -n 8 \
  --dist=worksteal --import-mode=importlib -p no:cacheprovider
=> 3990 passed, 8 skipped

python -m pytest -q tests/e2e/models/timm_seresnet/test_timm_seresnet_family_plugin.py
=> 14 passed

cmake --build $BUILD --target trtmc_model_timm_seresnet \
  test_timm_seresnet_image_preprocess_seam
$BUILD/test_timm_seresnet_image_preprocess_seam
=> build and link clean; test exit 0

python -m ruff check ... => All checks passed
python tools/legal_headers.py => findings=0
clang-format => clean
```

Numerical parity against timm's own implementation:

| Checkpoint | Correlation | argmax | top-5 |
| --- | --- | --- | --- |
| `timm/seresnet50.a1_in1k` | 0.99998800 | match | 5/5 |

The state dict loads into timm with no missing or unexpected keys.

### Hardware, Environment, and Revisions

- GPU: NVIDIA A100-SXM4-80GB, compute capability 8.0.
- Container: `Dockerfile.dev.x86` dev image, Ubuntu 24.04, Python 3.12.
- TensorRT 11.1.0.106, CUDA architecture `80-real`, Release build.
- Reference: timm 1.0.29 with torchvision 0.27.0+cpu on torch 2.12.0+cpu.
- Parity measured at fp32; the family also supports fp16.
- `timm/seresnet50.a1_in1k` @ `23f3482ee9acc4f51a2668a7cdf255fbd9420417`.

  The manifest does not pin `hf_revision`: the timm reference resolves
  `hf-hub:<id>` at `main`, so a pin disagrees with the cache the warm step
  populates and fails the offline reference run. See #1121.

### Not Run / Remaining Gaps

- No E2E harness run. The manifest is registered but was not executed here.
- Only `seresnet50` was verified numerically. The SE-ResNeXt grouped variants
  reuse the group derivation already exercised by `timm_resnet`, and the deeper
  SE-ResNets reuse the layout discovery, but **neither was downloaded or
  compared**.
- `legacy_seresnet*` matches the prefixes but has a different stem and gate
  placement; it is not verified and would likely need its own handling.
- No performance numbers. The benchmark row is registered but was not run.

## Notes For Future Readers

This is the fourth family here with a squeeze-excite gate, and the combinations
still differ: MobileNetV3 uses ReLU with hard-sigmoid, EfficientNet SiLU with
sigmoid, RegNet and this family ReLU with sigmoid. None of the differences
changes a tensor shape, so check against timm rather than assuming.

### Risk level

- [x] Low
- [ ] Medium
- [ ] High

Additive family. Existing families are untouched except for shared registration
points, all widened rather than redirected, and the full CPU suite passes.

### CODE REVIEW

**Comment by zhenshanx-nv** [comment]
Superseded by #1208, which implements this family in the post-cutover layout introduced by #1188. This branch targets `python/tensorrt_model_connect/families/`, which no longer exists, so it cannot be rebased.
